driver/src/orderbook/auction_data_reader.rs

```diff
@@ -47,17 +47,16 @@ impl AuctionDataReader {
     }
 
     fn apply_auction_data(&mut self, auction_elements: Vec<StableXAuctionElement>) {
-        self.apply_auction_elements_to_account_state(auction_elements.iter());
         // Workaround for borrow checker that would complain that `extend`
         // borrows self as mutable while at the same time the filter
         // closure borrows self as immutable because of using `self.index`.
         let index = self.index;
-        self.orders.extend(
-            auction_elements
-                .into_iter()
-                .filter(|x| x.in_auction(index) && x.order.sell_amount > 0)
-                .map(|auction_element| auction_element.order),
-        );
+        let auction_elements = auction_elements
+            .iter()
+            .filter(|x| x.in_auction(index) && x.order.sell_amount > 0);
+        self.apply_auction_elements_to_account_state(auction_elements.clone());
+        self.orders
+            .extend(auction_elements.map(|auction_element| auction_element.order.clone()));
```

Comment:
Should this be part of the diff logic instead?

Reply:
Im not sure I can tell what the meaning of this change is. Seems like a reordering of the same logic.

Reply:
I think it's still valuable to not serialize it into the instance file since it's not needed.